braid/_braid.c

```diff
@@ -2727,9 +2833,27 @@ _braid_FRefine(braid_Core   core,
    {
       _braid_CoreElt(core, r_space) = 0;
    }
-
+      
+   /* If periodic, add refinement to ensure the required number of points */
+   if (periodic)
+   {
+      f_gupper++;
```

Comment:
Should the incrementing and decrementing of f_gupper go inside the routine?

Reply:
No.  One is an index and the other is a count (num points).  I probably could reorganize a bit so this incrementing/decrementing isn't done here.  I wasn't crazy about it when I wrote it, but it was easier than introducing a new variable that represents the count.

Reply:
Ok, thanks for the clarification